support/event/events.go

```diff
@@ -0,0 +1,242 @@
+package event
+
+import (
+	"errors"
+	"os"
+	"runtime/debug"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/project-flogo/core/support/log"
+)
+
+const (
+	ENV_PUBLISH_AUDIT_EVENTS_KEY = "FLOGO_PUBLISH_AUDIT_EVENTS"
+)
+
+type EventListener interface {
```

Comment:
Yup, makes sense.

Reply:
I'm wondering about the need to put event-related packages as children of "support".

Reply:
There isn't much to the event package, it can be just event.go, right in support, no need for a package.

Reply:
@vijaynalawade @mellistibco thoughts?

Reply:
ignoring whether the amount of code warrants a package.  We can also consider the aesthetics:
```
event.Listener
event.RegisterListener
```
vs 
```
support.EventListener
support.RegisterEventListener
```

Reply:
@fm-tibco : I would prefer event.Listener/event.RegisterListener as app eventing is first class in OSS. In fact, we should make it more verbose like `appevent` etc.

Reply:
+1 with @vijaynalawade : I think it should be an *event* package and it is already coded like that.

My main concern is about the import path. I don't understand why there is a *support* directory leading to import paths such as *github.com/project-flogo/core/support/log* or **github.com/project-flogo/core/support/event*.

Such packages should either live in their own path or IMHO in a subpath of the *engine* (*github.com/project-flogo/core/engine/event* for instance).

Reply:
@debovema the reason I placed it in support is because I believe it's a part of non functional feature of core. But I would like to know what @fm-tibco @mellistibco and @vijaynalawade  think about it ?

Reply:
I don't believe log or event should be top level packages, they function in a sort of "support" capacity at this moment and is not really core functionality/construct.  As for the packages living under engine, I'd rather not have an activity/trigger require a dependency on the engine package just for logging.  Maybe an alternative package name?

Reply:
@fm-tibco : I am not shocked of *log* being a top level package. If there is the possibility of this package being used without other base features of the *core* repository, it could even be moved to a separate Go module / Git repository. But it surely is a little overkill :)
For the *event* package, I think it can live in the *engine* because even though there is no compile dependency between packages using events and the *engine* package, on a runtime point-of-view they are bound.

Reply:
@debovema, you make a good point, so I'm fine with moving events under engine and leaving log under support.  @vijaynalawade, @mellistibco thoughts?

Reply:
+1 i like having events under the engine package, seems to make a lot of sense...